src/Trace/TraceClient.php

```diff
@@ -0,0 +1,185 @@
+     *           Console.
+     *     @type int $retries Number of retries for a failed request.
+     *           **Defaults to** `3`.
+     *     @type array $scopes Scopes to be used for the request.
+     * }
+     */
+    public function __construct(array $config = [])
+    {
+        if (!isset($config['scopes'])) {
+            $config['scopes'] = [self::FULL_CONTROL_SCOPE];
+        }
+
+        $this->connection = new Rest($this->configureAuthentication($config));
+    }
+
+    /**
+     * Sends a Trace log in a simple fashion.
+     *
+     * @see https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects/patchTraces
+     *
+     * @param  Trace $trace The trace log to send.
+     * @param array $options [optional] Configuration Options
+     * @return bool
+     * @throws ServiceException
+     */
+    public function insert(Trace $trace, array $options = [])
+    {
+        return $this->insertBatch([$trace], $options);
+    }
+
+    /**
+     * Sends multiple Trace logs in a simple fashion.
+     *
+     * @see https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects/patchTraces
+     *
+     * @param Trace[] $traces The trace logs to send.
+     * @param array $options [optional] Configuration Options
+     * @return bool
+     * @throws ServiceException
+     */
+    public function insertBatch(array $traces, array $options = [])
+    {
+        // throws ServiceException on failure
+        $this->connection->patchTraces([
+            'projectId' => $this->projectId,
+            'traces' => array_map(function ($trace) {
+                return [
+                    'projectId' => $this->projectId,
+                    'traceId' => $trace->traceId(),
+                    'spans' => array_map(function ($span) {
+                        return $span->info();
+                    }, $trace->spans())
+                ];
+            }, $traces)
+        ] + $options);
+        return true;
+    }
+
+    /**
+     * Lazily find or instantiates a trace. There are no network requests made at this
+     * point. To see the operations that can be performed on a trace please
+     * see {@see Google\Cloud\Trace\Trace}.
+
+     * @param  string $traceId [optional] The trace id of the trace to reference.
+     * @return Trace
+     */
+    public function trace($traceId = null)
+    {
+        return new Trace($this->connection, $this->projectId, $traceId);
+    }
+
+    /**
+     * Fetch all traces in the project
+     *
+     * @see https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces/list Trace list API documentation.
+     *
+     * @param array $options [optional] {
+     *      Configuration options.
+     *
+     *      @type string $viewType Type of data returned for traces in the list.
+     *            Can be one of 'VIEW_TYPE_UNSPECIFIED', 'MINIMAL', 'ROOTSPAN', or
+     *            'COMPLETE'
+     *      @type int $pageSize Maximum number of traces to return
```

Comment:
Can we note that this is per page, so as to not be confused with the `resultLimit` option?